libs/utils/analysis/frequency_analysis.py

```diff
@@ -447,6 +510,96 @@ def plotClusterFrequencyResidency(self, clusters=None,
 
         self._plotFrequencyResidency(residencies, 'cluster', xmax, pct, active)
 
+    def plotFrequencyTransitions(self, clusters=None, pct=False):
+        """
+        Plot frequency transitions count of a given cluster.
+
+        Requires cpu_frequency events to be available in the trace.
+
+        :param clusters: name of the clusters to be plotted (all of them by
+            default)
+        :type clusters: str ot list(str)
+
+        :param pct: plot frequency transitions in percentage
+        :type pct: bool
+        """
+        if not self._trace.hasEvents('cpu_frequency'):
+            self._log.warn('Events [cpu_frequency] not found, plot DISABLED!')
+            return
+
+        # Sanitize clusters
+        if clusters is None:
+            if not self._platform or 'clusters' not in self._platform:
+                self._log.warn('No platform data available, plot DISABLED!')
+                return
+            _clusters = self._platform['clusters'].keys()
+        else:
+            _clusters = listify(clusters)
+
+        n_plots = len(_clusters)
+        gs = gridspec.GridSpec(n_plots, 1)
+        fig = plt.figure()
+
+        # Precompute frequency transitions
+        transitions = {}
+        xmax = 0
+        for c in _clusters:
+            t = self._dfg_frequency_transitions(c)
+            if t is None:
+                self._log.warn('No data available for cluster {}, '
+                               'plot DISABLED!'.format(c))
+                return
+
+            if pct:
+                tot = t.transitions.sum()
+                t = t.apply(lambda x: x * 100.0 / tot)
+
+            transitions[c] = t
+            max_cnt = t.transitions.max()
+            if xmax < max_cnt:
+                xmax = max_cnt
+
+        yrange = 3 * n_plots
+        for idx, c in enumerate(_clusters):
+            t = transitions[c]
+
+            axes = fig.add_subplot(gs[idx])
+            if pct:
```

Comment:
Could we keep the behaviour of having one bar per frequency with the `pct=True` ? I'm getting staggered bars on a single plot, which makes it hard to figure out the percentage for all frequencies past the first one.